### PR TITLE
tslint to eslint

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **feat**

This PR resolves []

### Description

moving from tslint to eslint!

Todo:
- [ ] find an alternative rule for our custom `jsx-boolean-value`, or rewrite it for eslint.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- tslint to eslint

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (tslint to eslint)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e9aed1daab648b5a32e2a3802023a1d2a22b36ab
</p>

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  settings: {
+    react: { version: "detect" },
+    "import/parsers": { "@typescript-eslint/parser": [".ts", ".tsx"] },
+    "import/resolver": {
+      typescript: {
+        alwaysTryTypes: true,
+        project: "./tsconfig.json",
+      },
+    },
+  },
```
Comment (damassi) [inline .eslintrc.js:19]: I would suggest removing this. In a large project, adding in "full" typescript type-checking support will slow down linting to the point where its unusable. And this feature isn't needed to gain all of the benefits. ([This](https://github.com/artsy/force/blob/main/.eslintrc.js#L5) is what we use over in force.)

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  overrides: [
+    {
+      // only for typescript
+      files: ["*.ts", "*.tsx"],
+      parserOptions: { project: ["./tsconfig.json"] },
+    },
+  ],
```
Comment (damassi) [inline .eslintrc.js:26]: I would suggest removing this. In a large project, adding in "full" typescript type-checking support will slow down linting to the point where its unusable. And this feature isn't needed to gain all of the benefits. ([This](https://github.com/artsy/force/blob/main/.eslintrc.js#L5) is what we use over in force.)

**Comment by MounirDhahri** [comment]
Closing PR for inactivity - please feel free to reopen it if needed 👌

**Pull request closed without merging**